module/localisation/BallLocalisation/src/BallLocalisation.cpp

```diff
@@ -79,25 +96,45 @@ namespace module::localisation {
         /* To run whenever a ball has been detected */
         on<Trigger<message::vision::Balls>, With<FieldDescription>>().then(
             [this](const message::vision::Balls& balls, const FieldDescription& field) {
-                if (balls.balls.size() > 0) {
-                    /* Call Time Update first */
-                    using namespace std::chrono;
-                    const auto curr_time  = NUClear::clock::now();
-                    const double seconds  = duration_cast<duration<double>>(curr_time - last_time_update_time).count();
-                    last_time_update_time = curr_time;
-                    filter.time(seconds);
-
-                    /* Now call Measurement Update. Supports multiple measurement methods
-                     * and will treat them as
-                     * separate measurements */
-                    for (auto& measurement : balls.balls[0].measurements) {
-                        filter.measure(Eigen::Vector3d(measurement.rBCc.cast<double>()),
-                                       Eigen::Matrix3d(measurement.covariance.cast<double>()),
-                                       field,
-                                       balls.Hcw);
+                // Call Time Update first
+                using namespace std::chrono;
+                const auto curr_time  = NUClear::clock::now();
+                const double seconds  = duration_cast<duration<double>>(curr_time - last_time_update_time).count();
+                last_time_update_time = curr_time;
+                filter.time(seconds);
+                for (const auto& ball : balls.balls) {
+                    if (!ball.measurements.empty()) {
+
+                        // Now call Measurement Update. Supports multiple measurement methods and will treat them as
+                        // separate measurements
+                        for (auto& measurement : ball.measurements) {
+                            filter.measure(Eigen::Vector3d(measurement.srBCc.cast<double>()),
+                                           Eigen::Matrix3d(measurement.covariance.cast<double>()),
+                                           field,
+                                           balls.Hcw);
+                        }
+                        last_measurement_update_time = curr_time;
                     }
-                    last_measurement_update_time = curr_time;
                 }
             });
+
+        on<Trigger<ResetBallHypotheses>, With<Sensors>, With<Field>, With<FieldDescription>, Sync<BallLocalisation>>()
```

Comment:
https://github.com/NUbots/NUbook/issues/120

Probs beyond just adding additional comments to explain what's going on because I fundamentally don't understand what hypotheses.